### PR TITLE
Lagrangian diagnostics : mean and interpolation

### PR DESCRIPTION
**Pull request by zcasparco**

### CODE REVIEW

**Comment by zcasparco** [comment]
Il ne devrait pas y avoir de conflit, à l'exception peut-être de parcels.py dans lequel  j'ai rajouté une fonction de conversion des unité de degré/s à m/s .

File: mitequinox/parcels.py [hunk 1]
```diff
@@ -1481,3 +1481,12 @@ def plot_h3_simple(
     )
     plt.xticks([], [])
     plt.yticks([], [])
+
+    
+
+# ------------------------------ unit converters ---------------------------------------
+
+def degs2ms(df):
```
Comment (apatlpo) [inline mitequinox/parcels.py:1489]: Add keyword argument with default value and `lat="lat"` use the argument when calling latitude (e.g. `df[lat_key]`)

Add documentation with `Parameters` section

Add link to parcels code in documentation:
https://github.com/OceanParcels/parcels/blob/6c83aa0acfe74f89bb8ce7a01de6357329d488d4/parcels/tools/converters.py

**Review by apatlpo — CHANGES_REQUESTED** [review]
please rename notebooks:
- `Interpolation_trajectories.ipynb` to `mean_field_interpolation.ipynb`
- `Mean_std_flts.ipynb` to `binned_statistics.ipynb`

In `binned_statistics.ipynb`:

- modify title: 
    - delete "Notebook :" (we know it's one ...).
    - add "binned" somewhere
- add value of `dl` parameter in in diagnostic file name. Typically: `L_dl1_mean_std.nc`
- add attribute with parcels simulation name in dataset
- add number of points used in each bin as an additional variable

In `mean_field_interpolation.ipynb`:

- delete first cell (`cluster.close()`)
- delete useless library imports
- improve code lisibility: cell 9 (`interp_trajectories`) introducing new variables like `U_mean`, `V_mean` is often not necessary, best to keep variables in xarray datasets
- cell 20, plotting all faces one by one is not super efficient, consider using the `quick_llc_plot` or just plotting one cell
- there are obvious issues with the statistics in the lower part of the notebook. Do you know what's going on?
- `add_geodata` needs to be moved to `parcels.py`

notebooks `parcels/h3.ipynb` and `overview.ipynb` have been modified accidently. 
The modifications are minor but you want to avoid such modifications in a PR.

Try also to avoid adding or deleteing new lines, etc ... (`mitequinox/parcels.py`, `mitequinox/utils.py`)

**Comment by apatlpo** [comment]
@zcasparco si tu veux que je regarde à nouveau ce PR, il faut que tu me le dises explicitemen

**Comment by apatlpo** [comment]
@zcasparco je viens de regarder le résultat de l'interpolation et ai quelques petites suggestions:

- Si tu part de l'archive parquet "time", le nom de l'achive avec les champs moyens interpolés devrait être "time_mean" 
- Il faudrait faire une archive interpolée pour chaque type d'indexation: "time", "trajectory"
- les noms des variables devraient être plus explicites et coller avec ceux existants:
    - SSU_interp -> mean_zonal_velocity
    - SSV_interp -> mean_meridional_velocity
    - Eta_interp -> mean_sea_level
    - SSS_interp -> mean_salinity
    - SST_interp -> mean_temperature

**Comment by apatlpo** [comment]
Other comments specific to `binned_diags.ipynb`:

- try to delete useless library imports
- try also to avoid import of the following type: `from library import *` (this makes code very difficult to understand/debug)
- `df_uv = df.drop(columns=[c for c in df.columns[6:9].append(df.columns[11:15]).values])` ??? why not `df_uv = df.drop(columns=["sea_level", ...])` ?
- `for V in df_uv.columns[4:-2]:` ?? labels are here to avoid such things: `for V in ["zonal_velocity", ...]: `
- some lines are two long, try to break and indent such that code is easily readable
- `df_mean.attrs = {'run_name': run_name,'dl': dl}` ... unless attrs dict does not exists, this is very dangerous as it could delete existing attributes. `df_mean.attrs.update(**{'run_name': run_name,'dl': dl})`
- some clean up at the end of the notebook seems necessary

**Comment by apatlpo** [comment]
@zcasparco what's the status on this?
Can we merge?

**Comment by zcasparco** [comment]
Almost, I just have to clean a few figures and it should be good

**Comment by apatlpo** [comment]
good to go @zcasparco ?

**Comment by zcasparco** [comment]
@apatlpo Yes I think it is good to go.

**Pull request merged**